pkg/handler/iam_handler_test.go

```diff
@@ -0,0 +1,565 @@
+	}{
+		{
+			name: "success",
+			orgsServer: &fakeOrgsServer{
+				policy: &iampb.Policy{
+					Bindings: []*iampb.Binding{
+						// Non-AOD bindings.
+						{
+							Members: []string{
+								"user:test-org-userA@example.com",
+							},
+							Role: "roles/accessapproval.approver",
+						},
+						// Unexpired bindings to be de-dupped.
+						{
+							Members: []string{
+								"user:test-org-userB@example.com",
+							},
+							Role: "roles/accessapproval.approver",
+							Condition: &expr.Expr{
+								Title:      ConditionTitle,
+								Expression: fmt.Sprintf("request.time < timestamp('%s')", time.Now().Add(1*time.Hour).Format(time.RFC3339)),
+							},
+						},
+						// Unexpired bindings to be de-dupped.
+						{
+							Members: []string{
+								"user:test-org-userA@example.com",
+								"user:test-org-userC@example.com",
+							},
+							Role: "roles/accessapproval.approver",
+							Condition: &expr.Expr{
+								Title:      ConditionTitle,
+								Expression: fmt.Sprintf("request.time < timestamp('%s')", time.Now().Add(1*time.Hour).Format(time.RFC3339)),
+							},
+						},
+					},
+				},
+			},
+			foldersServer: &fakeFoldersServer{
+				policy: &iampb.Policy{},
+			},
+			projectsServer: &fakeProjectsServer{
+				policy: &iampb.Policy{},
+			},
+			request: &v1alpha1.IAMRequestWrapper{
+				IAMRequest: &v1alpha1.IAMRequest{
+					ResourcePolicies: []*v1alpha1.ResourcePolicy{
+						{
+							Resource: "organizations/foo",
+							Bindings: []*v1alpha1.Binding{
+								{
+									Members: []string{
+										"user:test-org-userA@example.com",
+										"user:test-org-userB@example.com",
+									},
+									Role: "roles/accessapproval.approver",
+								},
+							},
+						},
+						{
+							Resource: "folders/bar",
+							Bindings: []*v1alpha1.Binding{
+								{
+									Members: []string{
+										"user:test-folder-user@example.com",
+									},
+									Role: "roles/cloudkms.cryptoOperator",
+								},
+							},
+						},
+						{
+							Resource: "projects/baz",
+							Bindings: []*v1alpha1.Binding{
+								{
+									Members: []string{
+										"user:test-project-user@example.com",
+									},
+									Role: "roles/bigquery.dataViewer",
+								},
+							},
+						},
+					},
+				},
+				Duration: 2 * time.Hour,
+			},
+			wantPolicies: []*v1alpha1.IAMResponse{
+				{
+					Resource: "organizations/foo",
+					Policy: &iampb.Policy{
+						Bindings: []*iampb.Binding{
+							{
+								Members: []string{
+									"user:test-org-userA@example.com",
+								},
+								Role: "roles/accessapproval.approver",
+							},
+							{
+								Members: []string{
+									"user:test-org-userC@example.com",
+								},
+								Role: "roles/accessapproval.approver",
+								Condition: &expr.Expr{
+									Title:      ConditionTitle,
+									Expression: fmt.Sprintf("request.time < timestamp('%s')", time.Now().Add(1*time.Hour).Format(time.RFC3339)),
```

Comment:
This is where flakiness will happen. `time.Now()` here could be different from `time.Now()` called above. Need to use a fixed value created before the test cases.